File: CPP/Stack_Dynamic.cpp
#include <iostream>
using namespace std;

void createList(int size) {
    int localList[size]; 

    for (int i = 0; i < size; i++)
        localList[i] = i;

    cout << "Stack Dynamic List: ";
    for (int i = 0; i < size; i++)
        cout << localList[i] << " ";
    cout << endl;
}

int main() {
    createList(5);
    createList(8);
    return 0;
}

File: Python/Stack_Dynamic.py

def create_list(size):
    # Local list, created dynamically inside function (stack allocation)
    local_list = [i for i in range(size)]
    print(f"Stack Dynamic List inside function: {local_list}")
    return local_list

if __name__ == "__main__":
    create_list(5)
    create_list(10)
